connexion/security.py

```diff
@@ -45,6 +94,87 @@ class SecurityHandlerFactory:
     context_kw = "context_"
     client = None
 
+    # TODO: Extract into separate class? (That has a security_handler_factory attribute?)
+    def parse_security_scheme(
```

Comment:
Idea is that the SecurityHandlerFactory can get subclasses and this function overwritten:
```python
def parse_security_scheme(...):
    if security_scheme["type"] == "oidc":
        # oidc implementation
    return super().parse_security_scheme(...)
```

Reply:
I was thinking more in the lines of having a separate class for each type like we have with the validators. Something like:

```
security_handler_dict = {
    "oauth2": Oauth2Handler,
    "basic": BasicHandler,
    ...
}

def parse_security_scheme(...):
    handler = security_handler_dict.get("security_scheme["type"])
    return handler.get_fn(security_scheme, required_scopes)
```

and then move the `verify_<type>` functions from this class into those handlers as well.

This splits the functionality nicely per type and allows people to contribute their handlers back to connexion.

Wdyt?